File: internal/infra/aws/sns_client.go
package infra_aws

import (
	"github.com/aws/aws-sdk-go-v2/service/sns"
)

type SNSClient struct {
	*sns.Client
}

func NewSNSClient(region AwsRegion, stage AwsStage) *SNSClient {
	if stage != StageLocal {
		return &SNSClient{
			Client: sns.NewFromConfig(NewConfig(region)),
		}
	}

	return &SNSClient{
		Client: sns.NewFromConfig(NewLocalConfig(region)),
	}
}

File: src/infra/aws/sns_client/sns_client.ts
import { SNSClient } from "@aws-sdk/client-sns"
import { AwsRegion, AwsStage, newAwsConfig } from "../utils"

export class SNSClientWrapper extends SNSClient {
	constructor(region: AwsRegion, stage: AwsStage) {
		const config = newAwsConfig(region, stage)
		super(config)
	}
}